presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java

```diff
@@ -2624,10 +2720,20 @@ private void doInsertIntoNewPartition(HiveStorageFormat storageFormat, SchemaTab
             stagingPathRoot = getStagingPathRoot(insertTableHandle);
             ConnectorPageSink sink = pageSinkProvider.createPageSink(transaction.getTransactionHandle(), session, insertTableHandle);
             sink.appendPage(CREATE_TABLE_PARTITIONED_DATA_2ND.toPage());
-            getFutureValue(sink.finish());
+            Collection<Slice> fragments = getFutureValue(sink.finish());
+            metadata.finishInsert(session, insertTableHandle, fragments);
 
             // verify we did not modify the table directory
-            assertEquals(listAllDataFiles(transaction, tableName.getSchemaName(), tableName.getTableName()), existingFiles);
+            // TODO: After finish insert data is becoming visible in current transaction.
+            // TODO: Is this correct behaviour? Should it be consistent with inserting into existing table/partition?
```

Comment:
What about this?

Reply:
It was kinda a weird behavioral. Metadata updates when inserting into an unpartitioned table are visible from within transaction, but table data is not. When for unpartitioned tables both are visible. But since https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java#L1830 it was intentional, and it is not a part of this PR to fix that.